analyzer/runtime/evm/erc721.go

```diff
@@ -62,6 +84,58 @@ func evmDownloadTokenERC721(ctx context.Context, logger *log.Logger, source node
 	return &tokenData, nil
 }
 
+func evmDownloadNFTERC721(ctx context.Context, logger *log.Logger, source nodeapi.RuntimeApiLite, ipfsClient ipfsclient.Client, round uint64, tokenEthAddr []byte, id *big.Int) (*EVMNFTData, error) {
+	var nftData EVMNFTData
+	supportsMetadata, err := detectInterface(ctx, logger, source, round, tokenEthAddr, ERC721MetadataInterfaceID)
+	if err != nil {
+		return nil, fmt.Errorf("checking ERC721Metadata interface: %w", err)
+	}
+	//nolint:nestif
+	if supportsMetadata {
+		if err = evmCallWithABI(ctx, source, round, tokenEthAddr, evmabi.ERC721Metadata, &nftData.MetadataURI, "tokenURI", id); err != nil {
+			if !errors.Is(err, EVMDeterministicError{}) {
+				return nil, fmt.Errorf("calling tokenURI: %w", err)
+			}
+			logDeterministicError(logger, round, tokenEthAddr, "ERC721Metadata", "tokenURI", err,
```

Comment:
Nit: Can you add "deterministically" to the log message inside this helper func, and/or some other reassuring phrase? Right now if I saw that log, I'd be a little alarmed and felt like I needed to investigate.

Reply:
ya adding

Reply:
added